x/wasm/keeper/submsg_test.go

```diff
@@ -652,6 +652,29 @@ func TestMigrateGovSubMsgAuthzPropagated(t *testing.T) {
 
 	var instanceLevel int
 	// mock wasvm to return new migrate msgs with the response
+	mockWasmVM.MigrateWithInfoFn = func(codeID wasmvm.Checksum, env wasmvmtypes.Env, migrateMsg []byte, migrateInfo wasmvmtypes.MigrateInfo, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) (*wasmvmtypes.ContractResult, uint64, error) {
+		if instanceLevel == 1 {
+			return &wasmvmtypes.ContractResult{Ok: &wasmvmtypes.Response{}}, 0, nil
+		}
+		instanceLevel++
+		submsgPayload := fmt.Sprintf(`{"sub":%d}`, instanceLevel)
+		return &wasmvmtypes.ContractResult{
+			Ok: &wasmvmtypes.Response{
+				Messages: []wasmvmtypes.SubMsg{
+					{
+						ReplyOn: wasmvmtypes.ReplyNever,
+						Msg: wasmvmtypes.CosmosMsg{
+							Wasm: &wasmvmtypes.WasmMsg{Migrate: &wasmvmtypes.MigrateMsg{
+								ContractAddr: example1.Contract.String(),
+								NewCodeID:    example2.CodeID,
+								Msg:          []byte(submsgPayload),
+							}},
+						},
+					},
+				},
+			},
+		}, 0, nil
```

Comment:
nit: Could you deduplicate this? This code and the one for MigrateFn are identical.

Reply:
https://github.com/CosmWasm/wasmd/pull/2022